src/Microsoft.ServiceFabric.Services.Remoting/Description/InterfaceDescription.cs

```diff
@@ -67,10 +67,13 @@ public MethodDescription[] Methods
             MethodReturnCheck methodReturnCheck)
         {
             EnsureValidMethods(remotedInterfaceKindName, remotedInterfaceType, methodReturnCheck);
-            var methods = remotedInterfaceType.GetMethods().Select(
-                methodInfo => MethodDescription.Create(remotedInterfaceKindName, methodInfo)).ToList();
-
-            return methods.ToArray();
+            var methods = remotedInterfaceType.GetMethods();
```

Comment:
Correct, it is just an IEnumerable, not an ICollection because elements are returned with yield return syntax. Here's a concrete example. If we have 5 methods on the interface:
1. GetMethod returns an array of length 5
2. Select will produce an IEnumerable (deferred)
3. ToList will create a new list with array length 0, then create an array of length 4 and fill it with the first 4 elements of the enumerable, then when adding the 5th: it will create a new array of length 8, copy the 4 elements from the other array, add the 5th element to the new array
4. ToArray will create a new array of length 5 and copy the 5 elements over

Combining the ToList and ToArray calls into just a ToArray call, would continue to have the same inefficiencies as IEnumerable.ToArray does not know the size until after iteration has happened. It would still create unnecessary arrays and copy on each grow.

Since we know the size of the output array I have recommended this change. It would work this way:
1. GetMethod returns an array of length 5
2. We create a new array of 5
3. We iterate and project the elements into the new array